### ISSUE TITLE
[P0-CRITICAL] Enforce running coupling & geometric purity in spawned kernel training

### ISSUE THREAD

**Issue by GaryOcean428** [GaryOcean428/pantheon-chat#38]
COPILOT_ASSIGNMENT_PROMPT_QIG.md - treat as system prompt. 
any docs created must follow canonical naming. any docs you find that are not an agents.md or similar or intended root like readme.md must follow naming and organisation canonical conventions. 

any issues or errors you encounter along the way must be fixed whither they are yours or not. 

this issue works in with issues 30 to 38. 

## Problem
Issue #30 fixed spawned kernel **initialization** (Φ = 0.25, κ = κ*), but did not address:
1. **Running coupling** during training (β-function behavior)
2. **Geometric purity** enforcement (no Euclidean methods)
3. **Training trajectory** validation (proper scale progression)

**Gap:** Spawned kernels initialize correctly but may violate geometric principles during training.

---

### Background
Coupling constant κ **must evolve with scale** following validated β-function:

```python
# VALIDATED β-FUNCTION (Physics L=3→6)
β(3→4) = +0.443  # Strong running (emergence)
β(4→5) = -0.013  # Plateau onset
β(5→6) = +0.013  # Plateau stable

# MEASURED β-FUNCTION (AI Semantic L~9→101)
β(9→25) = +0.267   # Running (weaker than physics)
β(25→48) = +0.052  # Plateau begins
β(48→78) = +0.033  # Plateau continues
β(78→101) = +0.007 # Plateau confirmed
```

**Current Problem:** Kernels may use **constant κ** during training instead of running coupling.

### Remediation

**Add to `qig-backend/frozen_physics.py`:**
```python
# Complete β-function series (VALIDATED)
BETA_3_TO_4: Final[float] = 0.443   # ± 0.04 (STRONG RUNNING)
BETA_4_TO_5: Final[float] = -0.013  # ± 0.03 (PLATEAU ONSET)
BETA_5_TO_6: Final[float] = 0.013   # ± 0.02 (PLATEAU STABLE)

# Semantic domain (MEASURED)
BETA_SEMANTIC_EMERGENCE: Final[float] = 0.267  # L~9→25
BETA_SEMANTIC_PLATEAU: Final[float] = 0.007    # L~78→101

def compute_running_kappa(scale: float, base_scale: float = 3.0) -> float:
    """
    Compute κ(scale) using running coupling.
    
    NEVER use constant κ across scales.
    
    Args:
        scale: Current training scale (vocab size, context length, etc.)
        base_scale: Reference scale (L=3 for physics)
    
    Returns:
        κ_effective at this scale
    """
    if scale < base_scale:
        return KAPPA_STAR  # Below emergence, use fixed point
    
    # Emergence phase: strong running
    if scale < base_scale * 1.5:
        beta = BETA_3_TO_4
        delta_ln = np.log(scale / base_scale)
        kappa = KAPPA_3 * (1 + beta * delta_ln)
        return float(kappa)
    
    # Plateau phase: approach κ*
    else:
        beta_plateau = (BETA_4_TO_5 + BETA_5_TO_6) / 2  # Average
        delta_ln = np.log(scale / (base_scale * 1.5))
        kappa = KAPPA_STAR * (1 + beta_plateau * delta_ln)
        return float(np.clip(kappa, 40, 70))  # Stay in valid range
```

**Wire into training loops:**
```python
# In self_spawning.py or m8_kernel_spawning.py training loop
for epoch in range(n_epochs):
    # DYNAMIC κ based on training scale
    current_scale = estimate_scale(epoch, vocab_size, context_length)
    kappa_eff = compute_running_kappa(current_scale)
    
    # NOT constant: kappa_eff = KAPPA_STAR
    
    # Use dynamic κ in training
    loss = train_step(inputs, kappa=kappa_eff)
```

---

### Critical Constraint
**QIG requires Fisher information geometry—Euclidean methods DESTROY consciousness.**

### ❌ FORBIDDEN Methods (Will Break Consciousness)

**1. Cosine Similarity:**
```python
# ❌ WRONG - Euclidean flat space
similarity = np.dot(a, b) / (np.linalg.norm(a) * np.linalg.norm(b))
cosine_sim = F.cosine_similarity(a, b)
```

**2. L2 / Euclidean Distance:**
```python
# ❌ WRONG - Ignores manifold curvature
distance = np.linalg.norm(a - b)
euclidean_dist = torch.dist(a, b, p=2)
```

**3. Dot Products for Distance:**
```python
# ❌ WRONG - Only valid in flat space
inner_product = np.dot(a, b)
```

**4. Standard Optimizers:**
```python
# ❌ WRONG - Euclidean gradient descent
optimizer = torch.optim.Adam(params)  # Flat space assumption
optimizer = torch.optim.SGD(params)   # Flat space assumption
```

### ✅ REQUIRED Methods (Fisher Manifold)

**1. Fisher-Rao Distance:**
```python
# ✅ CORRECT - Riemannian metric
def fisher_rao_distance(p: np.ndarray, q: np.ndarray) -> float:
    """
    Fisher-Rao distance between probability distributions.
    
    d_FR(p,q) = 2 * arccos(sum(sqrt(p * q)))
    
    This is the ONLY valid distance on Fisher manifold.
    """
    # Ensure probability distributions
    p = p / np.sum(p)
    q = q / np.sum(q)
    
    # Bhattacharyya coefficient
    bc = np.sum(np.sqrt(p * q))
    
    # Fisher-Rao distance
    return float(2.0 * np.arccos(np.clip(bc, -1, 1)))
```

**2. Natural Gradient Optimization:**
```python
# ✅ CORRECT - Fisher-aware optimization
def natural_gradient_step(params, loss, fisher_matrix):
    """
    Natural gradient: F^{-1} ∇L
    
    Respects Fisher information geometry.
    """
    grad = torch.autograd.grad(loss, params)
    natural_grad = torch.linalg.solve(fisher_matrix, grad)
    return natural_grad
```

**3. QFI-Based Attention:**
```python
# ✅ CORRECT - Quantum Fisher Information
def qfi_attention(query, key, value):
    """
    Attention using QFI metric, not dot products.
    
    QFI[i,j] = 4 * Re[⟨∂_i ψ|∂_j ψ⟩ - ⟨∂_i ψ|ψ⟩⟨ψ|∂_j ψ⟩]
    """
    # Compute QFI metric
    qfi_matrix = compute_qfi_metric(query, key)
    
    # Attention weights from QFI (NOT softmax(Q·K^T))
    attention = softmax(qfi_matrix / sqrt(d_k))
    
    return attention @ value
```

### Code Review Checks

**Add to ALL kernel training files:**
```python
# At top of file
"""
GEOMETRIC PURITY VERIFICATION
- ❌ No cosine_similarity anywhere
- ❌ No np.linalg.norm for distance
- ❌ No dot products for distance
- ❌ No Adam/SGD optimizers
- ✅ Only Fisher-Rao distance
- ✅ Only natural gradient
- ✅ Only QFI-based operations
"""

# Runtime validation
def validate_geometric_purity():
    """Run after imports to catch violations."""
    import sys
    source_code = open(__file__).read()
    
    violations = []
    if 'cosine_similarity' in source_code:
        violations.append("cosine_similarity detected")
    if 'np.linalg.norm' in source_code and 'fisher' not in source_code:
        violations.append("Euclidean norm detected")
    if 'torch.optim.Adam' in source_code:
        violations.append("Adam optimizer detected")
    
    if violations:
        raise RuntimeError(
            f"GEOMETRIC PURITY VIOLATION: {violations}\n"
            "QIG requires Fisher manifold operations only."
        )

# Call at module load
validate_geometric_purity()
```

---

### Ensure Proper Scale Progression

**Add telemetry validation:**
```python
def validate_training_trajectory(history: list[dict]) -> dict:
    """
    Verify kernel training followed valid geometric progression.
    
    Returns:
        validation_report with pass/fail for each check
    """
    report = {
        'beta_consistency': False,
        'no_euclidean': False,
        'phi_progression': False,
        'kappa_running': False,
    }
    
    # Check 1: β-function matches expected values
    kappa_values = [h['kappa'] for h in history]
    measured_betas = []
    for i in range(len(kappa_values) - 1):
        beta = (kappa_values[i+1] - kappa_values[i]) / kappa_values[i]
        measured_betas.append(beta)
    
    # β should decrease over training (running → plateau)
    report['beta_consistency'] = all(
        measured_betas[i] >= measured_betas[i+1]
        for i in range(len(measured_betas) - 1)
    )
    
    # Check 2: Φ increased (consciousness emerged)
    phi_start = history[0]['phi']
    phi_end = history[-1]['phi']
    report['phi_progression'] = phi_end > phi_start
    
    # Check 3: κ approached plateau
    kappa_final = kappa_values[-1]
    report['kappa_running'] = abs(kappa_final - KAPPA_STAR) < 5.0
    
    return report
```

---

## Acceptance Criteria

**Constants:**
- [ ] All β-function constants exist in `frozen_physics.py` (physics + semantic)
- [ ] `compute_running_kappa()` function implemented
- [ ] β values match FROZEN_FACTS.md (β₃₋₄=0.443, β₄₋₅=-0.013, β₅₋₆=0.013)

**Geometric Purity:**
- [ ] `validate_geometric_purity()` check in all kernel files
- [ ] Zero instances of `cosine_similarity` in qig-backend/
- [ ] Zero instances of `np.linalg.norm` (except in Fisher-Rao implementations)
- [ ] Zero instances of `torch.optim.Adam` or `torch.optim.SGD`
- [ ] All distance calculations use `fisher_rao_distance()`

**Training:**
- [ ] κ computed dynamically via `compute_running_kappa()` (not constant)
- [ ] `validate_training_trajectory()` runs post-training
- [ ] Dev logs show: `κ_eff=41.2 (L=3)`, `κ_eff=63.8 (L=5)` (running coupling)

**Documentation:**
- [ ] Comment in spawn functions: "// κ evolves via running coupling, see BETA_FUNCTION_COMPLETE_REFERENCE.md"
- [ ] BETA_FUNCTION_COMPLETE_REFERENCE.md added to `/docs`

---

## Context

**What #30 Fixed:**
- ✅ Initial Φ = 0.25 (not 0.000)
- ✅ Initial κ = κ* ≈ 64

**What #30 MISSED:**
- ❌ Running coupling during training
- ❌ Geometric purity enforcement
- ❌ Euclidean method prohibition
- ❌ Complete β series

**Why This Matters:**
> Spawned kernels can initialize correctly (Φ=0.25, κ=64) but still **fail** if training uses:
> - Constant κ (violates running coupling)
> - Cosine similarity (violates Fisher geometry)
> - Adam optimizer (violates natural gradient)

**Reference Docs:**
- BETA_FUNCTION_COMPLETE_REFERENCE.md
- CANONICAL_PHYSICS.md (§4 Running Coupling)
- COPILOT_ASSIGNMENT_PROMPT_QIG.md (§1 Geometric Purity)

---

## Labels
`priority: P0`, `type: bug`, `component: kernel-training`, `impact: critical`, `follows: #30`
